### ISSUE TITLE
Exit from "minishift ssh" console returns error

### ISSUE THREAD

**Issue by agajdosi** [minishift/minishift#660]
**Environment:**
OS: RHEL 7.3 Workstation+Server, Windows 7, Fedora 24
Minishift: _1.0.0-beta.5
CDK: 3.0.0-beta.3

**Description:**
I connect to console of minishift machine with `minishift ssh` command. When I exit this ssh instance and last command executed here returned error, then also this exit will throw an error:
```
[hudson@localhost ~]$ minishift ssh
Last login: Tue Apr  4 09:24:43 2017 from 192.168.42.1
[docker@minishift ~]$ mistake
-bash: mistake: command not found
[docker@minishift ~]$ exit
logout
E0404 09:31:11.723209    5180 ssh.go:38] Cannot establish SSH connection to the VM:  exit status 127
```
This is just a cosmetic as I did not find any problem after - for example I am able to connect again with `minishift ssh`.

**Comment by jorgemoralespou** [comment]
Same here on mac and minishift 1.0.0.rc.1

**Comment by hferentschik** [comment]
I think this a case of bad error message and in fact bad error handling. The error message originates here -https://github.com/minishift/minishift/blob/7974f5edc9697c6b878794aa9f98d9d0fe668588/cmd/minishift/cmd/ssh.go#L39

Obviously the message is misleading, since we actually had a connection. The other problem is that per default the ssh session returns with the exit command of the last command executed prior exiting the the ssh session. In case of typing a unknown command the exit code ($?) will be 127 and hence the error we are seeing here. 

If possible we need to distinguish between not being able to make the ssh connection (in which case the current message would make sense) and having an actual session which ends with a non 0 exit code. I would think having an interactive session one should not report any error based on the commands run on the VM.

**Comment by jhcook** [comment]
I have no idea if this is related, but I am having the same issue, and it appears it is an erroneous use of a volume. The / is filling up and I can no longer connect. 

https://github.com/minishift/minishift/issues/355

**Comment by gbraad** [comment]
@jhcook your message is unrelated, as this happens because of the SSH daemon stopping to work.

**Comment by gbraad** [comment]
```
$ minishift ssh "mount.nfs -v"
usage: mount.nfs remotetarget dir [-rvVwfnsh] [-o nfsoptions]
options:
        -r              Mount file system readonly
        -v              Verbose
        -V              Print version
        -w              Mount file system read-write
        -f              Fake mount, do not actually mount
        -n              Do not update /etc/mtab
        -s              Tolerate sloppy mount options rather than fail
        -h              Print this help
        nfsoptions      Refer to mount.nfs(8) or nfs(5)

mount.nfs: no mount point provided
Cannot establish SSH connection to the VM: exit status 32
```

**Comment by gbraad** [comment]
Minikube does not handle the error besides logging it:
```
			glog.Errorln(errors.Wrap(err, "Error attempting to ssh/run-ssh-command"))
			os.Exit(1)
```

Docker-machine does the following:
```
		Action:          runCommand(cmdSSH),

func cmdSSH(c CommandLine, api libmachine.API) error {
       // ...

	client, err := host.CreateSSHClient()
	if err != nil {
		return err
	}

	return client.Shell(c.Args().Tail()...)
}
```

**Comment by gbraad** [comment]
quick fix... drop the:
```
atexit.ExitWithMessage(1, fmt.Sprintf("Cannot establish SSH connection to the VM: %s", err.Error()))
```
and do
```
os.Exit(0); // or os.Exit(1)
```

**Comment by hferentschik** [comment]
I am basically ok with _os.Exit(0), even though since this is the last command it would not even be needed. 

That said, if one looks into the implementation of CreateSSHShell, there are several checks performed prior to creating the shell. Errors in these checks would get lost, even though they are the ones for which the atExit makes most sense. As discussed in this issue, the last command of the interactive sessions should not play a role.

If exceptions sub-classes would be properly used, one could inspect the returned error, but I believe the only distinguishing factor are the error messages. Building code around this is not really a good idea. So I suggest to add the additional checks from _CreateSSHShell_ to the ssh command. We do similar checks in the other commands, so it would be consistent. If we make these checks explicitly we can then just ignore the error returned by _cluster.CreateSSHShell_. _os.Exit_ is then not even needed.

**Comment by hferentschik** [comment]
Merged via pull request #1198
